test/extensions/issuanceExtension.spec.ts

```diff
@@ -0,0 +1,733 @@
+    managerRedeemFee = ether(.03);
+    feeRecipient = delegatedManager.address;
+    managerIssuanceHook = ADDRESS_ZERO;
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  describe("#constructor", async () => {
+    let subjectManagerCore: Address;
+    let subjectIssuanceModule: Address;
+
+    beforeEach(async () => {
+      subjectManagerCore = managerCore.address;
+      subjectIssuanceModule = issuanceModule.address;
+    });
+
+    async function subject(): Promise<IssuanceExtension> {
+      return await deployer.globalExtensions.deployIssuanceExtension(
+        subjectManagerCore,
+        subjectIssuanceModule
+      );
+    }
+
+    it("should set the correct IssuanceModule address", async () => {
+      const issuanceExtension = await subject();
+
+      const storedModule = await issuanceExtension.issuanceModule();
+      expect(storedModule).to.eq(subjectIssuanceModule);
+    });
+  });
+
+  describe("#initializeModule", async () => {
+    let subjectDelegatedManager: Address;
+    let subjectCaller: Account;
+    let subjectMaxManagerFee: BigNumber;
+    let subjectManagerIssueFee: BigNumber;
+    let subjectManagerRedeemFee: BigNumber;
+    let subjectFeeRecipient: Address;
+    let subjectManagerIssuanceHook: Address;
+
+    beforeEach(async () => {
+      await issuanceExtension.connect(owner.wallet).initializeExtension(delegatedManager.address);
+
+      subjectDelegatedManager = delegatedManager.address;
+      subjectCaller = owner;
+      subjectMaxManagerFee = maxManagerFee;
+      subjectManagerIssueFee = managerIssueFee;
+      subjectManagerRedeemFee = managerRedeemFee;
+      subjectFeeRecipient = feeRecipient;
+      subjectManagerIssuanceHook = managerIssuanceHook;
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return issuanceExtension.connect(subjectCaller.wallet).initializeModule(
+        subjectDelegatedManager,
+        subjectMaxManagerFee,
+        subjectManagerIssueFee,
+        subjectManagerRedeemFee,
+        subjectFeeRecipient,
+        subjectManagerIssuanceHook
+      );
+    }
+
+    it("should correctly initialize the IssuanceModule on the SetToken", async () => {
+      await subject();
+
+      const isModuleInitialized: Boolean = await setToken.isInitializedModule(issuanceModule.address);
+      expect(isModuleInitialized).to.eq(true);
+
+      const storedSettings: any = await issuanceModule.issuanceSettings(setToken.address);
+
+      expect(storedSettings.maxManagerFee).to.eq(maxManagerFee);
+      expect(storedSettings.managerIssueFee).to.eq(managerIssueFee);
+      expect(storedSettings.managerRedeemFee).to.eq(managerRedeemFee);
+      expect(storedSettings.feeRecipient).to.eq(feeRecipient);
+      expect(storedSettings.managerIssuanceHook).to.eq(managerIssuanceHook);
+    });
+
+    it("should emit the correct ModuleInitialized event", async () => {
+      await expect(subject()).to.emit(setToken, "ModuleInitialized").withArgs(issuanceModule.address);
```

Comment:
I wouldn't worry about checking the event. We only want to test that the params were passed to the module correctly which the previous test does. The event doesn't really tell us anything about if params were correctly submitted.

Reply:
This goes for all the extension tests